### DISCUSSION TITLE
Troubleshooting Batch Rendering

### DISCUSSION THREAD

**Discussion by batman4444** [pixijs/pixijs#9642] [Q&A]
I am working on a Pixi based game and am trying to troubleshoot some performance issues I am running into on mobile devices. I have installed Spector.js and after some digging I suspect most of the sprites in my project are not batch rendering. I apologize in advance, I know very little about this kind of stuff, but I have been poking around online for numerous hours online now and I need some help.

How do I confirm if batch rendering is correctly happening? In Spector.js in the left column (where the frame buffer steps are displayed) I see that all of the shadows for my sprites are all being drawn in a single frame buffer step. I assume this means that the shadows correctly got batched. However, later I see each one of my sprites individually being added to the scene one at a time each on their own frame buffer step. I am assuming this means no batching is happening for the majority of the sprites.

I am using the Pixi Projection plug in and the sprites that do not appear to be batching are all children of a Projection Container3d. I am concerned this might be causing the issue? Can anyone confirm this is/isn't the issue?  

If it is the Projection plugin causing the issue, is there a way to manually set up the scene (like assigning batching groups) to take advantage of batching while still using the Container3d? Are there any tutorials/articles that explain all this?

Some details about my project: Using Pixi verson 6.5.8, pixi-layers and pixi-projection. Many of the sprites share the exact same texture and they all come from the same spritesheet. I already culled offscreen sprites and “flattened” the sprite textures (I was initially layering them) and I did see some improvement in total commands/draws

**Comment by batman4444** [comment]
It appears my issue was being caused by applying a ColorMatrixFilter to one of the children sprites.  If I remove the filter, things seem to batch up as expected. I went from hundreds of draw calls to under 10

**Reply by miltoncandelero** [reply]
Yes, filters will break batching. However, tint won't and it's free. Feel free to try with that if you need some color effects
